plugins/pencilblue/templates/angular/admin/content/articles/article_form.html

```diff
@@ -35,6 +36,21 @@
 			});
 		};
 
+		$scope.generateUrl = function() {
+			$scope.article.url = $scope.article.headline.replace(/\s+/g, '-').toLowerCase();
+			if($scope.article.url){
+				$scope.urlGenerated = true;
+			} else {
+				$scope.urlGenerated = false;
+			}
+		};
+
+		$scope.$watch('article.url', function() {
```

Comment:
```$scope.$watch('article.url', function(newVal, oldVal)...```

Then you can check if the newVal is different than the oldVal before doing a string compare on the generated URL.